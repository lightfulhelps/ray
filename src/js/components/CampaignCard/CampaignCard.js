// @flow
import * as React from 'react';
import classNames from 'classnames';

type Props = {
  className?: string,
  cover: string,
  status: string,
  tag?: string,
  title: string,
};

const CampaignCard = ({ className, cover, status, tag: Tag = 'div', title, ...other }: Props) => {
  const classes = classNames(
    className,
    'campaign-card',
    'card rounded-lg shadow text-decoration-none h-100'
  );
  const statusClasses = classNames(
    'text-xs font-weight-bold',
    { 'text-primary': status === 'Live' },
    { 'text-warning': status === 'Draft' }
  );

  return (
    <Tag {...other} className={classes} data-test-id="campaign-card">
      <div
        className="campaign-card-cover"
        data-test-id="campaign-card-cover"
        style={{
          backgroundImage: `url(${cover})`,
        }}
      />
      <div className="px-2 py-1 h-100 d-flex flex-column justify-content-between">
        <div className="h6" data-test-id="campaign-card-title">
          {title}
        </div>
        <div className={statusClasses} data-test-id="campaign-card-status">
          {status}
        </div>
      </div>
    </Tag>
  );
};

export default CampaignCard;
