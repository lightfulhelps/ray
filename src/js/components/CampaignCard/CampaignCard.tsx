import * as React from 'react';
import classNames from 'classnames';

type Props = {
  className?: string;
  cover: string;
  state: string;
  tag?: keyof JSX.IntrinsicElements;
  title: string;
  [key: string]: any;
};

const CampaignCard: React.FC<Props> = ({
  className,
  cover,
  state = '',
  tag: Tag = 'div',
  title,
  ...other
}: Props) => {
  const classes = classNames(
    className,
    'campaign-card',
    'card rounded-lg shadow text-decoration-none h-100'
  );
  const stateClasses = classNames(
    'text-xs font-weight-bold',
    { 'text-primary': state.toLowerCase() === 'live' },
    { 'text-warning': state.toLowerCase() === 'draft' }
  );

  return (
    <Tag {...other} className={classes} data-test-id="campaign-card">
      <div
        className="campaign-card-cover"
        data-test-id="campaign-card-cover"
        style={{
          backgroundImage: `url(${cover})`,
        }}
      />
      <div className="px-2 py-1 h-100 d-flex flex-column justify-content-between">
        <div className="h6" data-test-id="campaign-card-title">
          {title}
        </div>
        <div className={stateClasses} data-test-id="campaign-card-state">
          {state}
        </div>
      </div>
    </Tag>
  );
};

export default CampaignCard;
