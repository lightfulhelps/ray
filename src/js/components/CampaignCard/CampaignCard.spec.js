import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';
import CampaignCard from './CampaignCard';

const setup = (overrides = {}) => {
  const props = merge(
    {
      cover: 'http://lorempixel.com/640/480',
      title: 'THX distributed',
      state: 'Live',
    },
    overrides
  );
  const wrapper = shallow(<CampaignCard {...props} />);

  return { wrapper, props };
};

describe('<CampaignCard />', () => {
  it('should render', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle className', () => {
    const { wrapper } = setup();

    wrapper.setProps({ className: 'custom' });

    expect(wrapper.hasClass('campaign-card')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });

  it('should handle the cover prop', () => {
    const cover = 'http://lorempixel.com/b924f7e6-5434-4ef4-bdc6-2670b799f27c';
    const { wrapper } = setup({
      cover,
    });

    expect(
      wrapper.find('[data-test-id="campaign-card-cover"]').prop('style').backgroundImage
    ).toContain(cover);
  });

  it('should handle the title prop', () => {
    const title = 'azure Handcrafted Wooden Car';
    const { wrapper } = setup({ title });

    expect(wrapper.find('[data-test-id="campaign-card-title"]').text()).toEqual(title);
  });

  it('should handle the Live state', () => {
    const { wrapper } = setup({ state: 'Live' });

    expect(wrapper.find('[data-test-id="campaign-card-state"]').text()).toEqual('Live');

    expect(wrapper.find('[data-test-id="campaign-card-state"]').prop('className')).toContain(
      'text-primary'
    );

    expect(wrapper.find('[data-test-id="campaign-card-state"]').prop('className')).not.toContain(
      'text-warning'
    );
  });

  it('should handle the Draft state', () => {
    const { wrapper } = setup({ state: 'Draft' });

    expect(wrapper.find('[data-test-id="campaign-card-state"]').text()).toEqual('Draft');

    expect(wrapper.find('[data-test-id="campaign-card-state"]').prop('className')).not.toContain(
      'text-primary'
    );

    expect(wrapper.find('[data-test-id="campaign-card-state"]').prop('className')).toContain(
      'text-warning'
    );
  });

  it('should handle the tag prop', () => {
    const { wrapper } = setup();

    expect(wrapper.type()).toBe('div');

    wrapper.setProps({ tag: 'a' });

    expect(wrapper.type()).toBe('a');
  });

  it('should pass through other props', () => {
    const { wrapper } = setup({ tabIndex: 1, id: 'test' });

    expect(wrapper.prop('tabIndex')).toEqual(1);
    expect(wrapper.prop('id')).toEqual('test');
  });
});
