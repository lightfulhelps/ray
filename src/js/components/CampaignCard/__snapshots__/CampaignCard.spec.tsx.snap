// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CampaignCard /> should render 1`] = `
<div
  className="campaign-card card rounded-lg shadow text-decoration-none h-100"
  data-test-id="campaign-card"
>
  <div
    className="campaign-card-cover"
    data-test-id="campaign-card-cover"
    style={
      Object {
        "backgroundImage": "url(http://lorempixel.com/640/480)",
      }
    }
  />
  <div
    className="px-2 py-1 h-100 d-flex flex-column justify-content-between"
  >
    <div
      className="h6"
      data-test-id="campaign-card-title"
    >
      THX distributed
    </div>
    <div
      className="text-xs text-primary"
      data-test-id="campaign-card-state"
    >
      Live
    </div>
  </div>
</div>
`;
